api/v1beta1/cloudstackmachine_webhook.go

```diff
@@ -55,18 +57,25 @@ var _ webhook.Validator = &CloudStackMachine{}
 func (r *CloudStackMachine) ValidateCreate() error {
 	cloudstackmachinelog.Info("validate create", "name", r.Name)
 
-	var (
-		errorList field.ErrorList
-		spec      = r.Spec
-	)
+	var errorList field.ErrorList
 
 	// IdentityRefs must be Secrets.
-	if spec.IdentityRef != nil && spec.IdentityRef.Kind != defaultIdentityRefKind {
+	if r.Spec.IdentityRef != nil && r.Spec.IdentityRef.Kind != defaultIdentityRefKind {
 		errorList = append(errorList, field.Forbidden(field.NewPath("spec", "identityRef", "kind"), "must be a Secret"))
 	}
 
-	errorList = webhook_utilities.EnsureFieldExists(spec.Offering, "Offering", errorList)
-	errorList = webhook_utilities.EnsureFieldExists(spec.Template, "Template", errorList)
+	affinity := strings.ToLower(r.Spec.Affinity)
+	if !(affinity == "" || affinity == "no" || affinity == "pro" || affinity == "anti") {
```

Comment:
Can we just define enums for these instead of using strings? https://www.sohamkamani.com/golang/enums/#enum-types

Reply:
Here's an attempted start branch as PR-ed against this one. https://github.com/aws/cluster-api-provider-cloudstack/pull/26/files

Enums in GO really stink, and you can kind of see why in the linked branch.

On top of all the weird extra code to make a type act like an enum, the code still doesn't reject the assignment of a value that's not in the enum's set. So I tried creating a cluster with `Affinity: pr` instead of `pro` and it happily accepted it.

Writing an enum is like writing an entire class for a field value.